### DISCUSSION TITLE
Has anyone considered running PostgREST as an AWS Lambda ?

### DISCUSSION THREAD

**Discussion by fusionbeam** [PostgREST/postgrest#2918] [Q&A]
AWS Aurora Serverless v2 begs the question: can PostgREST be run as an AWS Lambda? And would it be perform?

I am considering it for a very elastic pay-on-demand service in production that does not see schema changes too often.
AWS Lambda being a stateless service the question is how much state does Postgrest need? 
If we were to serialise the cached schema to a file and bundle it in the lambda runtime would that suffice?

Thanks,
Rares

**Comment by wolfgangwalther** [comment]
I have not, but I always wondered whether `postgrest` could also be used in a useful way as a library and not an executable. Just found this: https://theam.github.io/aws-lambda-haskell-runtime/. Maybe it's possible to use that to hook the postgrest core up to aws lambda.

Just a note though. I have no intention to even try myself.

**Reply by steve-chavez** [reply]
> Just a note though. I have no intention to even try myself.

Unfortunately I'm on the same boat for the meantime too. That being said I believe it would be useful to have library mode for postgrest. 

> If we were to serialise the cached schema to a file

That seems possible. It might even be few lines of code to read a JSON file with `Aeson` and our `SchemaCache` data type.

**Comment by emberdart** [comment]
This is something I've been thinking of, so I could use as a library (optionally caching the schema?), so I would import the Application-exposing module, extract the Application, construct a Request from the aeson-decoded input, run the Application on the request, take the IO Response to produce the output and shove it out. Similar to something crazy I started here when attempting to [link DigitalOcean's cloud functions to a generic `Request -> IO Response`](https://github.com/JolHarg/hs-openfaas/blob/main/lib/Network/DigitalOcean/CloudFunctions/Handler.hs) which I also added to Servant. Much more required but [proof of concept works for Discord webhook bots](https://github.com/JolHarg/hs-discord-webhook/blob/main/src/Main.hs).

**Reply by ikeboy003** [reply]
Hi,

https://github.com/ikeboy003/cloudrest I adapted postgrest to run on cloudflare workers

**Reply by wolfgangwalther** [reply]
> https://github.com/ikeboy003/cloudrest I adapted postgrest to run on cloudflare workers

A vibe-coded project that violates our copyright\* - great (watch out - sarcasm could be involved).

\* Our LICENSE terms have not been abided by, but at the same time it's easy to find very specific traces of the PostgREST source code in the repo.

**Reply by ikeboy003** [reply]
fair, that was missing, fixed it. thanks for the catch, and thanks for postgrest.

**Comment by mkleczek** [comment]
#5031 and #5028, once merged, will enable running PostgREST as AWS lambda function: mount S3 bucket as a filesystem, configure PostgREST to dump schema cache to a file on the mounted filesystem, and to load it at startup (using `file://` URI to point to a file).

**Reply by emberdart** [reply]
I wonder how easy that would be to port to digitalocean or faas in general
